strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-pypi-layout-provider/src/main/java/org/carlspring/strongbox/artifact/coordinates/PypiWheelArtifactCoordinates.java

```diff
@@ -0,0 +1,238 @@
+     * 
+     * @param distribution: Uniquely identifying artifact coordinate (required)
+     * @param version: Packages current version (required)
+     * @param build: Build_tag parameter (optional)
+     * @param lang_impl_version: Language and Implementation version argument (required)
+     * @param abi: ABI tag parameter (required)
+     * @param platform: Platform tag parameter (required)
+     * @return bar Returns bar
+     */
+    public PypiWheelArtifactCoordinates(String distribution,
+            String version,
+            String build,
+            String lang_impl_version,
+            String abi,
+            String platform)
+    {
+        // if any of the required arguments are empty, throw an error
+        if (distribution == "" || version == "" || lang_impl_version == "" || abi == "" || platform == "")
+        {
+            throw new IllegalArgumentException("distribution, version, language_implementation_version_tag, abi_tag, and platform_tag must be specified");
+        }
+
+        if (build != "" && !Character.isDigit(build.charAt(0)))
+        {
+            throw new IllegalArgumentException("illegal build tag");
+        }
+
+        setId(distribution);
+        setVersion(version);
+        setBuild(build);
+        setLang(lang_impl_version);
+        setAbi(abi);
+        setPlatform(platform);
+    }
+
+    /**
+     * @param path: The filename of the PyPi Wheel package
+     * @return Returns a PyPiWheelArtifactCoordinates object with all included coordinates set
+     */
+    public static PypiWheelArtifactCoordinates parse(String path)
+    {
+        return PypiWheelArtifactCoordinatesUtils.parse(path);
+    }
+
+    /**
+     * @return Returns distribution coordinate value (serves as the unique ID)
+     */
+    @Override
+    public String getId()
+    {
+        return getCoordinate(DISTRIBUTION);
+    }
+
+    /**
+     * @param id: DISTRIBUTION coordinate will take this value
+     */
+    @Override
+    public void setId(String id)
+    {
+        setCoordinate(DISTRIBUTION, id);
+    }
+
+    /**
+     * @return Returns the VERSION coordinate value
+     */
+    @Override
+    public String getVersion()
+    {
+        return getCoordinate(VERSION);
+    }
+
+    /**
+     * @param version: VERSION coordinate takes this value
+     */
+    @Override
+    public void setVersion(String version)
+    {
+        setCoordinate(VERSION, version);
+    }
+
+    /**
+     * @return Returns the BUILD_TAG coordinate value
+     */
+    @ArtifactLayoutCoordinate
+    public String getBuild()
+    {
+        return getCoordinate(BUILD_TAG);
+    }
+
+    /**
+     * @param build: BUILD_TAG coordinate will take this value
+     */
+    public void setBuild(String build)
+    {
+        setCoordinate(BUILD_TAG, build);
+    }
+
+    /**
+     * @return Returns the LANG_IMPL_VERSION_TAG coordinate value
+     */
+    @ArtifactLayoutCoordinate
+    public String getLang()
```

Comment:
Could you please rename this to `getLanguageImplementationVersion()`? Thanks! :)